server/service/redis_policy_set/redis_policy_set.go

```diff
@@ -48,18 +48,21 @@ func (r *redisFailingPolicySet) ListSets() ([]uint, error) {
 
 // AddHost adds the given host to the policy sets.
 func (r *redisFailingPolicySet) AddHost(policyID uint, host fleet.PolicySetHost) error {
-	conn := redis.ConfigureDoer(r.pool, r.pool.Get())
-	defer conn.Close()
-
 	// The order of the following two operations is important.
 	//
 	// The ordering of operations in AddHost and RemoveSet has been chosen to avoid
 	// ending up with a policySetKey with a host entry, but without its corresponding entry
 	// in the set of sets.
-	if _, err := conn.Do("SADD", policySetKey(policyID), hostEntry(host)); err != nil {
+	//
+	// The commands are run in different functions with different connections in
+	// case it is running in redis cluster mode - the keys may not live on the
+	// same node.  There is no additional connection cost for standalone mode, as
+	// the second connection will come from the pool (same for cluster mode if
+	// both key happen to live on the same node).
```

Comment:
Aren't all of the operations using the same key `policySetKey(policyID)` ?

Reply:
No, there's `policies:failing:{id}` for each policy, and `policies:failing_sets` (the set of sets, to avoid scanning the keyspace).

Reply:
Oh right right.